Configuring ED failed

I assume that it is caused by TF listeners in ED plugins not being handled correctly. I will put some effort in it today, 2017-13-06.
```
terminate called after throwing an instance of 'tf2::LookupException'
  what():  "map" passed to lookupTransform argument target_frame does not exist. 
/home/amigo/ros/kinetic/system/devel/lib/ed/ed(_Z13signalHandleri+0x558)[0x4dbd17]
/lib/x86_64-linux-gnu/libc.so.6(+0x354b0)[0x7f029a7754b0]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x38)[0x7f029a775428]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x16a)[0x7f029a77702a]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x16d)[0x7f029adaf84d]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8d6b6)[0x7f029adad6b6]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c6a9)[0x7f029adac6a9]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(__gxx_personality_v0+0x2e5)[0x7f029adad005]
/lib/x86_64-linux-gnu/libgcc_s.so.1(+0xff83)[0x7f029ab19f83]
/lib/x86_64-linux-gnu/libgcc_s.so.1(_Unwind_Resume+0x57)[0x7f029ab1a487]
[ERROR] [1512505793.347695]: Could not configure ED:

Service call failed: transport error completing service call: unable to receive data from sender, check sender's logs for details
[amigo/ed_configure-4] process has died [pid 11714, exit code 1, cmd /home/amigo/ros/kinetic/system/src/ed/tools/configure /home/amigo/ros/kinetic/system/src/amigo_bringup/parameters/world_modeling/world_model.yaml __name:=ed_configure __log:=/home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed_configure-4.log].
log file: /home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed_configure-4*.log
[amigo/ed-3] process has died [pid 11709, exit code 1, cmd /home/amigo/ros/kinetic/system/devel/lib/ed/ed __name:=ed __log:=/home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed-3.log].
log file: /home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed-3*.log
```

--- a/src/localization_plugin.cpp
+++ b/src/localization_plugin.cpp
@@ -115,7 +115,27 @@ void LocalizationPlugin::configure(tue::Configuration config)
 
         // Get homogeneous transformation between odom and base link frame
         tf::StampedTransform tf_odom_base_link;
-        tf_listener_->lookupTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), tf_odom_base_link);
+
+        //Set to zero, in case of error when looking up, because can't break an if loop in case of error.
+        tf_odom_base_link.frame_id_ = odom_frame_id_;
+        tf_odom_base_link.child_frame_id_ = base_link_frame_id_;
+        tf_odom_base_link.stamp_ = ros::Time::now();
+        tf_odom_base_link.setOrigin(tf::Vector3(0,0,0));
+        tf_odom_base_link.setRotation(tf::Quaternion(0,0,0,0));
+
+        if (tf_listener_->waitForTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), ros::Duration(5)))
+        {
+            try
+            {
+                tf_listener_->lookupTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), tf_odom_base_link);
+            }
+            catch (tf::TransformException ex)
+            {
+                ROS_ERROR("[ED Localisation] %s",ex.what());
+            }
+        }
+        else
+            ROS_ERROR("[ED Localisation] no transform between odom and base_link.");
 
         // Calculate base link position in map frame
         tf::Transform tf_map_base_link = homogtrans_map_odom*tf_odom_base_link;

--- a/src/localization_plugin.cpp
+++ b/src/localization_plugin.cpp
@@ -117,17 +117,21 @@ void LocalizationPlugin::configure(tue::Configuration config)
         tf::StampedTransform tf_odom_base_link;
 
         //Set to zero, in case of error when looking up, because can't break an if loop in case of error, so we need to proceed with empty transform.
-        tf_odom_base_link.frame_id_ = odom_frame_id_;
-        tf_odom_base_link.child_frame_id_ = base_link_frame_id_;
-        tf_odom_base_link.stamp_ = ros::Time::now();
-        tf_odom_base_link.setOrigin(tf::Vector3(0,0,0));
-        tf_odom_base_link.setRotation(tf::Quaternion(0,0,0,0));
-
-        if (tf_listener_->waitForTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), ros::Duration(5)))
+        if (tf_listener_->waitForTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), ros::Duration(1)))
         {
             try
             {
                 tf_listener_->lookupTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), tf_odom_base_link);
+                // Calculate base link position in map frame
+                tf::Transform tf_map_base_link = homogtrans_map_odom*tf_odom_base_link;
+                tf::Vector3 pos_map_baselink = tf_map_base_link.getOrigin(); // Returns a vector
+                tf::Quaternion rotation_map_baselink = tf_map_base_link.getRotation(); // Returns a quaternion
+
+                p.x = pos_map_baselink.x();
+                p.y = pos_map_baselink.y();
+                yaw = tf::getYaw(rotation_map_baselink);
+
+                ROS_DEBUG_STREAM("Initial pose from parameter server: [" << p.x << ", " << p.y << "], yaw:" << yaw);
             }
             catch (tf::TransformException ex)
             {
@@ -136,17 +140,6 @@ void LocalizationPlugin::configure(tue::Configuration config)
         }
         else
             ROS_ERROR("[ED Localisation] no transform between odom and base_link.");
-
-        // Calculate base link position in map frame
-        tf::Transform tf_map_base_link = homogtrans_map_odom*tf_odom_base_link;
-        tf::Vector3 pos_map_baselink = tf_map_base_link.getOrigin(); // Returns a vector
-        tf::Quaternion rotation_map_baselink = tf_map_base_link.getRotation(); // Returns a quaternion
-
-        p.x = pos_map_baselink.x();
-        p.y = pos_map_baselink.y();
-        yaw = tf::getYaw(rotation_map_baselink);
-
-        ROS_DEBUG_STREAM("Initial pose from parameter server: [" << p.x << ", " << p.y << "], yaw:" << yaw);
 
     }
     else if (config.readGroup("initial_pose", tue::OPTIONAL))

--- a/src/localization_plugin.cpp
+++ b/src/localization_plugin.cpp
@@ -106,7 +106,7 @@ void LocalizationPlugin::configure(tue::Configuration config)
     if (nh.getParam("initialpose", ros_param_position))
     {
 
-        //Make a homogeneous transformation with the variables from the parameter server
+        // Make a homogeneous transformation with the variables from the parameter server
         tf::Transform homogtrans_map_odom;
         homogtrans_map_odom.setOrigin(tf::Vector3(ros_param_position["x"], ros_param_position["y"], 0.0));
         tf::Quaternion q_map_odom;
@@ -116,14 +116,14 @@ void LocalizationPlugin::configure(tue::Configuration config)
         // Get homogeneous transformation between odom and base link frame
         tf::StampedTransform tf_odom_base_link;
 
-        //Set to zero, in case of error when looking up, because can't break an if loop in case of error, so we need to proceed with empty transform.
+        // Set to zero, in case of error when looking up, because can't break an if loop in case of error, so we need to proceed with empty transform.
         if (tf_listener_->waitForTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), ros::Duration(1)))
         {
             try
             {
                 tf_listener_->lookupTransform(odom_frame_id_, base_link_frame_id_, ros::Time(0), tf_odom_base_link);
                 // Calculate base link position in map frame
-                tf::Transform tf_map_base_link = homogtrans_map_odom*tf_odom_base_link;
+                tf::Transform tf_map_base_link = homogtrans_map_odom * tf_odom_base_link;
                 tf::Vector3 pos_map_baselink = tf_map_base_link.getOrigin(); // Returns a vector
                 tf::Quaternion rotation_map_baselink = tf_map_base_link.getRotation(); // Returns a quaternion
 
@@ -135,11 +135,11 @@ void LocalizationPlugin::configure(tue::Configuration config)
             }
             catch (tf::TransformException ex)
             {
-                ROS_ERROR("[ED Localisation] %s",ex.what());
+                ROS_ERROR("[ED Localization] %s",ex.what());
             }
         }
         else
-            ROS_ERROR("[ED Localisation] no transform between odom and base_link.");
+            ROS_ERROR("[ED Localization] no transform between odom and base_link.");
 
     }
     else if (config.readGroup("initial_pose", tue::OPTIONAL))
@@ -151,7 +151,7 @@ void LocalizationPlugin::configure(tue::Configuration config)
     }
 
     particle_filter_.initUniform(p - geo::Vec2(0.3, 0.3), p + geo::Vec2(0.3, 0.3), 0.05,
-                                    yaw - 0.1, yaw + 0.1, 0.05);
+                                 yaw - 0.1, yaw + 0.1, 0.05);
 
     config.value("robot_name", robot_name_);
 

--- a/src/localization_plugin.cpp
+++ b/src/localization_plugin.cpp
@@ -116,7 +116,7 @@ void LocalizationPlugin::configure(tue::Configuration config)
         // Get homogeneous transformation between odom and base link frame
         tf::StampedTransform tf_odom_base_link;
 
-        //Set to zero, in case of error when looking up, because can't break an if loop in case of error.
+        //Set to zero, in case of error when looking up, because can't break an if loop in case of error, so we need to proceed with empty transform.
         tf_odom_base_link.frame_id_ = odom_frame_id_;
         tf_odom_base_link.child_frame_id_ = base_link_frame_id_;
         tf_odom_base_link.stamp_ = ros::Time::now();

